### PR TITLE
Updated dashboard tab routes

### LINKED ISSUE #94: ROLL-52 : URL Route Updates
Make use of the URL paths available globally in app-routing.module.ts to update page routes. The focus in this issue to restructure the /home sub-route so that corresponding routes match.

### PR DESCRIPTION
**Pull request by brandlxee11**

### CODE REVIEW

File: src/app/pages/home/home-routing.module.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { HomePage } from './home.page';
 const routes: Routes = [
 
 {
-    path:'home',
+    path:'',
```
Comment (brickgriff) [inline src/app/pages/home/home-routing.module.ts]: Undo this change. It reads like there are two empty paths. This establishes `/home` for its child nodes: `/home/rolodex`, `/home/profile`, and `/home/exchange`. The empty path below (line 43) directs the user to `/home/rolodex` and the default path for `/home/` (line 36) leads to `rolodex` (which joins the first part to become `/home/rolodex` as well).

**Review by brickgriff — CHANGES_REQUESTED** [review]
Additionally, move the child path for profile up. As it, currently it looks like it's a link, like the one leading to `card-import`. It should look more like the other two pages (`/home/rolodex` and `/home/exchange`). There's no need for this to look like a child of itself.

https://github.com/bgriffiniv/rollme/blob/8fbf36759a245f7ce3024ca94b5d8b001422973e/src/app/pages/home/home-routing.module.ts#L24-L27

File: src/app/pages/home/home-routing.module.ts (hunk 1 above)
Comment (brandlxee11) [reply src/app/pages/home/home-routing.module.ts]: Two empty paths? But the URLs already have `/home` in them. Admittedly, I don't quite understand why the other routes need to be there.

Comment (brickgriff) [reply src/app/pages/home/home-routing.module.ts]: Instead of this...
```
const routes: Routes = [

{
    path:'',
    component: HomePage,
    children: [
      {
        path:'rolodex', loadChildren:'./rolodex/rolodex.module#RolodexPageModule'
      },

      {
        path:'exchange', loadChildren:'./exchange/exchange.module#ExchangePageModule'
      },

      {
        path:'profile',
        children: [
          {
            path:'',
            loadChildren:'./profile/profile.module#ProfilePageModule'
          },
          {
            path:'card-import',
            loadChildren: '../card-import/card-import.module#CardImportPageModule'
          }
        ]
      },

      {
        path: '',
        redirectTo: 'rolodex',
        pathMatch: 'full'
      }
    ]
  },
  {
    path: '',
    redirectTo: 'home/rolodex',
    pathMatch: 'full'
  }
];
```
try this...
```
const routes: Routes = [

  {
    path:'rolodex', loadChildren:'./rolodex/rolodex.module#RolodexPageModule'
  },
  {
    path:'exchange', loadChildren:'./exchange/exchange.module#ExchangePageModule'
  },
  {
    path:'profile', loadChildren:'./profile/profile.module#ProfilePageModule',
    children: [
      {
        path:'card-import',
        loadChildren: '../card-import/card-import.module#CardImportPageModule'
      }
    ]
  },
  {
    path: '',
    redirectTo: 'rolodex',
    pathMatch: 'full'
  }
];
```

Comment (brickgriff) [reply src/app/pages/home/home-routing.module.ts]: Depending on where `card-import` sits (maybe it's been moved already)

Comment (brandlxee11) [reply src/app/pages/home/home-routing.module.ts]: Nope. Tabs bar disappears. I'm removing the global path in `app-routing-module.ts` per the suggestion of a tutorial. It seems this delegates the task of specifying the path to the `home-routing-module.ts`.

Comment (brickgriff) [reply src/app/pages/home/home-routing.module.ts]: Is there any way to merge them?

```
const routes: Routes = [

{
    path:'',
    redirectTo: 'home/rolodex',
    pathMatch: 'full',
    component: HomePage,
    children: [
      {
        path:'rolodex', loadChildren:'./rolodex/rolodex.module#RolodexPageModule'
      },

      {
        path:'exchange', loadChildren:'./exchange/exchange.module#ExchangePageModule'
      },

      {
        path:'profile',
        children: [
          {
            path:'',
            loadChildren:'./profile/profile.module#ProfilePageModule'
          },
          {
            path:'card-import',
            loadChildren: '../card-import/card-import.module#CardImportPageModule'
          }
        ]
      },

      {
        path: '',
        redirectTo: 'rolodex',
        pathMatch: 'full'
      }
    ]
  }
];
```

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ const routes: Routes = [
   { path: 'user', loadChildren: './pages/menu/users/user-details/user-details.module#UserDetailsPageModule' },
   { path: 'user/:id', loadChildren: './pages/menu/users/user-details/user-details.module#UserDetailsPageModule' },
 
-  { path: 'home', loadChildren: './pages/home/home.module#HomePageModule' },
+  { path: '', loadChildren: './pages/home/home.module#HomePageModule' },
```
Comment (brickgriff) [inline src/app/app-routing.module.ts]: Okay... definitely don't do this. For the same reason as before. We already have a blank path at line 5-9

**Review by brickgriff — CHANGES_REQUESTED** [review]
I still need to experiment locally but this change doesn't seem right.

File: src/app/app-routing.module.ts (hunk 2 above)
Comment (brandlxee11) [reply src/app/app-routing.module.ts]: Okay...well I'll change it back, but after that I'm not sure what to do.

**Comment by brandlxee11** [comment]
What's it gonna take to complete this PR?

File: src/app/pages/home/home-routing.module.ts (hunk 1 above)
Comment (brandlxee11) [reply src/app/pages/home/home-routing.module.ts]: This approach didn't work either.

**Comment by brickgriff** [comment]
Putting `home` back in the root of the home router module doesn't seem to break anything, even if all the pages are specified in the app routing module. ~I'm still investigating how this works. It seems like we don't actually use the nested route, rather we always use the "global" routes. I use "global" loosely because it seems like once inside the nested route, accessing routes defined in the app routing module doesn't seem to work. In other words, it seems like we need to specify all routes needed for a given domain (e.g. the home routing module) since they don't talk to each other (the app routing module is not a proper "parent" of the home routing module).~ In order to load root pages, they need to be referenced in the `AppRoutingModule`. For example, to get to a path called `/login`, an object with the name `login` and a `loadChildren` field is needed. Same for `/home`. Once in the `HomePageModule`, getting to the tab pages only requires a relative path (for example, `exchange` would load `/home/exchange`). To access the `exchange` page from outside the `HomePageModule` the full path is required (`/home/exchange`).

**Comment by brickgriff** [comment]
Adding the `home` path on the root route does re-introduce the duplicated `/home/home/...` referencing behavior that we want to avoid. This happens because the first `home` is provided by the route in the app routing module and the second is provided by the home routing module. The app routing module requires routing like this for two reasons: 1) the default blank path is already pointing to `/login` and 2) getting to the home page requires a valid path to be available in the app routing module or else the page can't be loaded.

This poses an interesting problem. If the root says `home`, then the path gets a duplicate (URLs in the code can still be written properly and Angular will navigate to the correct page) because the app routing module already has that path. If the root is blank, then there are two blank routes in the home routing module. The solution is to remove the second default path again for two reasons: 1) the global default path `/` already leads to `/login` from app routing module and 2) once the home page module is loaded, the default path `/home/` already leads to `/home/rolodex` from the first route object in the list.

**Review by brickgriff — APPROVED** [review]
I apologize for the agonizing on details but that's what's going to help us swim more freely in the future. If nothing else, I've learned some characteristics and use cases for Angular navigation which informed my changes. I aim to reduce duplication, ambiguity, and other not-so-best practices. Try this out locally (using http://localhost/8100 to play with the URL directly).
